### ISSUE TITLE
telnet process is killed too soon in t/telnet.t

### ISSUE THREAD

**Issue by jberger** [jhthorsen/mojo-ioloop-readwritefork#1]
It seems that OSX and probably most BSDs (from the state of CPANTesters) do not have a localhost telnet server running. It causes tests to fail. Perhaps this test can be improved to detect this situation or else simplified to not use telnet.

**Comment by jberger** [comment]
``` bash
Joels-MBP:Test joel$ cpanm -v Mojo::IOLoop::ReadWriteFork
cpanm (App::cpanminus) 1.7018 on perl 5.016003 built for darwin-2level
Work directory is /Users/joel/.cpanm/work/1419607016.69954
You have make /usr/bin/make
You have LWP 6.08
You have /usr/bin/tar: bsdtar 2.8.3 - libarchive 2.8.3
You have /usr/bin/unzip
Searching Mojo::IOLoop::ReadWriteFork on cpanmetadb ...
--> Working on Mojo::IOLoop::ReadWriteFork
Fetching http://www.cpan.org/authors/id/J/JH/JHTHORSEN/Mojo-IOLoop-ReadWriteFork-0.09.tar.gz ... OK
Unpacking Mojo-IOLoop-ReadWriteFork-0.09.tar.gz
x Mojo-IOLoop-ReadWriteFork-0.09/
x Mojo-IOLoop-ReadWriteFork-0.09/.ship.conf
x Mojo-IOLoop-ReadWriteFork-0.09/cpanfile
x Mojo-IOLoop-ReadWriteFork-0.09/Makefile.PL
x Mojo-IOLoop-ReadWriteFork-0.09/lib/
x Mojo-IOLoop-ReadWriteFork-0.09/lib/Mojo/
x Mojo-IOLoop-ReadWriteFork-0.09/lib/Mojo/IOLoop/
x Mojo-IOLoop-ReadWriteFork-0.09/lib/Mojo/IOLoop/ReadWriteFork.pm
x Mojo-IOLoop-ReadWriteFork-0.09/README
x Mojo-IOLoop-ReadWriteFork-0.09/MANIFEST
x Mojo-IOLoop-ReadWriteFork-0.09/.travis.yml
x Mojo-IOLoop-ReadWriteFork-0.09/Changes
x Mojo-IOLoop-ReadWriteFork-0.09/META.yml
x Mojo-IOLoop-ReadWriteFork-0.09/t/
x Mojo-IOLoop-ReadWriteFork-0.09/t/drain.t
x Mojo-IOLoop-ReadWriteFork-0.09/t/telnet.t
x Mojo-IOLoop-ReadWriteFork-0.09/t/callback.t
x Mojo-IOLoop-ReadWriteFork-0.09/t/close.t
x Mojo-IOLoop-ReadWriteFork-0.09/t/bash.t
x Mojo-IOLoop-ReadWriteFork-0.09/t/00-basic.t
x Mojo-IOLoop-ReadWriteFork-0.09/t/leaky-errno.t
x Mojo-IOLoop-ReadWriteFork-0.09/META.json
x Mojo-IOLoop-ReadWriteFork-0.09/example/
x Mojo-IOLoop-ReadWriteFork-0.09/example/tail.pl
Entering Mojo-IOLoop-ReadWriteFork-0.09
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.02)
Running Makefile.PL
Configuring Mojo-IOLoop-ReadWriteFork-0.09 ... Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Mojo::IOLoop::ReadWriteFork
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have Test::More 0.88 ... Yes (1.001009)
Checking if you have IO::Pty 1.10 ... Yes (1.12)
Checking if you have Test::Memory::Cycle 1.04 ... Yes (1.04)
Checking if you have Mojolicious 4.30 ... Yes (5.68)
Building and testing Mojo-IOLoop-ReadWriteFork-0.09 ... cp lib/Mojo/IOLoop/ReadWriteFork.pm blib/lib/Mojo/IOLoop/ReadWriteFork.pm
Manifying 1 pod document
PERL_DL_NONLAZY=1 "/Users/joel/perl5/perlbrew/perls/perl-5.16.3/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-basic.t ..... ok   
t/bash.t ......... ok   
t/callback.t ..... 1/10 Unhandled type: GLOB at /Users/joel/perl5/perlbrew/perls/perl-5.16.3/lib/site_perl/5.16.3/Devel/Cycle.pm line 107.
t/callback.t ..... ok     
t/close.t ........ ok   
t/drain.t ........ Unhandled type: GLOB at /Users/joel/perl5/perlbrew/perls/perl-5.16.3/lib/site_perl/5.16.3/Devel/Cycle.pm line 107.
t/drain.t ........ ok   
t/leaky-errno.t .. ok   
t/telnet.t ....... # client<<<(telnet> )
# client<<<(open localhost 60202
# 
# )
# client<<<(Trying ::1...
# )
t/telnet.t ....... 1/? 
#   Failed test 'Connected'
#   at t/telnet.t line 46.
#                   'telnet> open localhost 60202
# 
# Trying ::1...
# '
#     doesn't match '(?^:Connected)'

#   Failed test 'got echo'
#   at t/telnet.t line 47.
#                   'telnet> open localhost 60202
# 
# Trying ::1...
# '
#     doesn't match '(?^s:I heard you say:.*hey)'

#   Failed test 'got drain event'
#   at t/telnet.t line 48.
#          got: '0'
#     expected: '1'
# Looks like you failed 3 tests of 3.
t/telnet.t ....... Dubious, test returned 3 (wstat 768, 0x300)
Failed 3/3 subtests 

Test Summary Report
-------------------
t/telnet.t     (Wstat: 768 Tests: 3 Failed: 3)
  Failed tests:  1-3
  Non-zero exit status: 3
Files=7, Tests=25,  1 wallclock secs ( 0.03 usr  0.01 sys +  0.68 cusr  0.11 csys =  0.83 CPU)
Result: FAIL
Failed 1/7 test programs. 3/25 subtests failed.
make: *** [test_dynamic] Error 3
FAIL
! Installing Mojo::IOLoop::ReadWriteFork failed. See /Users/joel/.cpanm/work/1419607016.69954/build.log for details. Retry with --force to force install it.
```

**Comment by jhthorsen** [comment]
I've been wanting to get hold on a Mac to fix this, but I forgot to ask my coworker before Christmas.

It's on my todo list though.

You're wrong about the telnet server though. The unit test creates a socket which the "telnet" application should be able to connect to.

**Comment by jhthorsen** [comment]
Thanks for the bug report @jberger. Makes it easier to remember that I have go fix this :)

Merry Christmas and a happy new year!

**Comment by jberger** [comment]
And to you too!

BTW I have a mac now if it helps you with testing.

**Comment by jhthorsen** [comment]
@jberger: I forgot to reference this issue, but could you pull and test the master branch?

The change is https://github.com/jhthorsen/mojo-ioloop-readwritefork/compare/93d2abf...1aa3e06

**Comment by jhthorsen** [comment]
Fixed in 0.10.
